crates/miden-objects/src/transaction/tx_args.rs

```diff
@@ -187,6 +188,17 @@ impl TransactionArgs {
         self.advice_inputs.extend(AdviceInputs::default().with_map(new_elements));
     }
 
+    /// Adds the `signature` corresponding to `public_key` on `message` to the advice inputs' map.
+    ///
+    /// The advice inputs' map is extended with the following key:
+    ///
+    /// - hash(public_key, message) |-> signature.
+    pub fn add_signature(&mut self, public_key: PublicKey, message: Word, signature: Vec<Felt>) {
```

Comment:
I think this works for now, but this does imply that the only public key we can provide is the commitment to the Falcon public key. Maybe instead the method could be something like:
```rust
pub fn add_signature(
    &mut self,
    public_key: impl Into<Word>,
    message: Word,
    signature: Vec<Felt>,
)
```
Though this does imply that a public key can be uniquely converted into a word, which is something we've been trying to figure out in https://github.com/0xMiden/crypto/pull/475#discussion_r2280316387.

Reply:
Agreed, with the only nit being that it should be `public_key_hash` :slightly_smiling_face:.

Reply:
If it is `Into<Word>` it could still be a `public_key` - no? And then we'd so do something like:
```rust
let pub_key_hash: Word = public_key.into();
```